tikv: add configuration docs for I/O rate limiter (#5729)
pingcap/docs: tikv-configuration-file.md

+ Default value: 45% of the size of total system memory
+ Unit: KB|MB|GB

## storage.io-rate-limit

Configuration items related to the I/O rate limiter.

### `max-bytes-per-sec`

+ Maximum I/O bytes that this server can write to or read from disk (determined by mode) in one second. Internally it prefers throttling background operations over foreground ones. This value should be set to the disk's optimal IO bandwidth, e.g. maximum IO bandwidth specified by cloud disk vendors. When set to zero, disk IO operations are not limited.

> + Default value: `"0MB"`

Reviewer: Commented on the other PR, can we use empty value to denote no IO limiter, instead of "0MB"?

Reviewer 2: @tabokie Any update?

Author: @TomShawn Discussed offline, this change is not needed now.

+ Default value: 45% of the size of total system memory
+ Unit: KB|MB|GB

## storage.io-rate-limit

Configuration items related to the I/O rate limiter.

### `max-bytes-per-sec`

> + Maximum I/O bytes that this server can write to or read from disk (determined by mode) in one second. Internally it prefers throttling background operations over foreground ones. This value should be set to the disk's optimal IO bandwidth, e.g. maximum IO bandwidth specified by cloud disk vendors. When set to zero, disk IO operations are not limited.

Author: This is actually an important feature of the new limiter. Instead of indiscriminately limit all IO types, it will throttle background operations first (e.g. compaction, import).

> + Maximum I/O bytes that this server can write to or read from disk (determined by mode) in one second. Internally it prefers throttling background operations over foreground ones. This value should be set to the disk's optimal IO bandwidth, e.g. maximum IO bandwidth specified by cloud disk vendors. When set to zero, disk IO operations are not limited.

Reviewer 2: Thanks for explaining. I've modify my comment. PTAL again~

> + Limits the maximum I/O bytes that a server can write to or read from the disk (determined by the `mode` configuration item below) in one second. With this limit is reached, TiKV prefers throttling background operations over foreground ones. The value of this configuration item should be set to the disk's optimal I/O bandwidth, for example, the maximum I/O bandwidth specified by your cloud disk vendor. When this configuration value is set to zero, disk I/O operations are not limited.

Reviewer 2: ```suggestion
+ Limits the maximum I/O bytes that a server can write to or read from the disk (determined by the `mode` configuration item below) in one second. When this limit is reached, TiKV prefers throttling background operations over foreground ones. The value of this configuration item should be set to the disk's optimal I/O bandwidth, for example, the maximum I/O bandwidth specified by your cloud disk vendor. When this configuration value is set to zero, disk I/O operations are not limited.
```

+ Default value: `"0MB"`

### `mode`

> + Determine which types of IO operations are counted and restrained below threshold. Only write-only mode is supported for now.

Reviewer 2: Does the "threshold" refer to max-bytes-per-sec?
```suggestion
+ Determines which types of I/O operations are counted and restrained below the threshold. Only the write-only mode is supported for now.
```

Author: Yes, do we need to substitute it here?

Reviewer 2: Yes, to make it explicit and clear.

> + Optional value: `"write-only"`, `"read-only"`, `"all-io"`

Reviewer 2: If  the write-only mode is supported for now, what about removing `"read-only"` and `"all-io"` for now?
```suggestion
+ Value options: `"write-only"`, `"read-only"`, `"all-io"`
```

Author: I'm not sure. If so this configuration item doesn't make a lot of sense to exist. Might as well remove it? What do you think?

Author: And BTW the expression `Optional value` is used several times before in this document. Do we need to change them as well?

Reviewer 2: + If it affects the user behavior, this config should be documented. If `"read-only"` and `"all-io"` are currently not supported, we should remove them and add them back when available.
+ OK. I've deleted the suggested comment. Thanks for reminding~
